src/nested_pandas/nestedframe/utils.py

```diff
@@ -12,29 +12,56 @@ class NestingType(Enum):
     NESTED = "nested"
 
 
-def _expr_nesting_type(node: ast.expr | None) -> set[NestingType]:
+def _actionable_splits(parents: list[ast.expr], node: ast.expr | None) -> dict[str, list]:
+    """
+    Given an expression which contains references to both base and nested columns,
+    return a dictionary of the sub-expressions that should be evaluated independently.
+    The key of the dictionary is the name of the nested column, and will be a blank
+    string in the case of base columns.  The value is a list of the parent nodes
+    that lead to sub-expressions that can be evaluated successfully.
+
+    While this is not in use today for automatically splitting expressions, it can
+    be used to detect whether an expression is suitably structured for evaluation:
+    the returned dictionary should have a single key.
+    """
     if not isinstance(node, ast.expr):
-        return set()
+        return {}
     if isinstance(node, ast.Name):
-        return {NestingType.BASE}
-    if isinstance(node, ast.Attribute):
-        return {NestingType.NESTED}
+        return {"": parents}
+    if isinstance(node, ast.Attribute) and isinstance(node.value, ast.Name):
+        return {node.value.id: parents}
     sources = (
         [getattr(node, "left", None), getattr(node, "right", None)]
         + getattr(node, "values", [])
         + getattr(node, "comparators", [])
     )
-    result: set[NestingType] = set()
+    result: dict[str, list] = {}
     for s in sources:
-        result.update(_expr_nesting_type(s))
+        child = _actionable_splits(parents, s)
+        for k, v in child.items():
+            result[k] = result.get(k, []) + v
+    # After a complete traversal across sources, check for any necessary splits.
+    # If it's homogenous, move the split-node up the tree.
+    if len(result) == 1:
+        # Let the record of each parent node drift up the tree,
+        # and merge the subtrees into a single node, since by definition,
+        # this node is homogeneous over all of its children, and can
+        # be evaluated in a single step.
+        result = {k: [node] for k in result}
+    else:
+        # At this point, we need to split the expression.  The idea here is that
+        # we want a succession of efficient queries, each of which will produce
+        # a subset of either the base or the nested columns.
+        pass
```

Comment:
This is currently a no-op with confusing documentation. Do we still need this code block or was this a remnant from a different way of constructing the recursion?

Reply:
Good eye!  It is indeed a no-op, but the no-op is very intentional, hence the comment.  You're right, this was scaffolding for a more ambitious future implementation, which I scaled back.  I'll restructure it to lose the `else / pass`, and make that comment more pertinent to this more concise algorithm.